src/hooks/usePagination.js

```diff
@@ -1,14 +1,71 @@
-export const DOTS = "...";
+export const DOTS = "..."; //ellipsis
 
-function usePagination() {
-  /*
-    Rewrite the logic here to map out the pagination to be displayed
+/**
+ * usePagination: Custom hook that returns pagination data based on
+ * currently selected details:
+ * - Uses a set to avoid duplicates
+ * - Formats the set data to be used in the pagination component
+ * - Adds ellipsis to gaps (ie. non-siblings from the set)
+ * - returns an array of page numbers with placed ellispsis
+ */
 
-    !!!!!! ATTENTION !!!!!!
-    Please replace this comment here with a description of this hook.
-    
-  */
-  return [1, 2, 3, DOTS, 5];
+function usePagination({ currentPage, pageSize, totalCount }) {
```

Comment:
The value returned by this hook can be memoized/cached to avoid unnecessarily running the logic every time the hook is called. The useMemo hook seems suitable.